theano/configdefaults.py

```diff
@@ -1524,11 +1536,24 @@ def _timeout_default():
 
 try:
     p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    cxx_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    cxx_version_str = 'CXX_NOT_FOUND'
+
+try:
+    p_out = output_subprocess_Popen(['g++', '-dumpversion'])
     gcc_version_str = p_out[0].strip().decode()
 except OSError:
     # Typically means gcc cannot be found.
-    gcc_version_str = 'GCC_NOT_FOUND'
+    gcc_version_str = 'CXX_NOT_FOUND'
 
+try:
+    p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    icc_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    icc_version_str = 'CXX_NOT_FOUND'
```

Comment:
I would be ok with a single cxx_version_str.  Otherwise, it would be kind of crazy to have a plethora of version strings for every different type of compiler.